proc/proc.go

```diff
@@ -149,24 +149,25 @@ func GetKernelModules(modulesPath string,
 }
 
 // ListPIDs from the proc filesystem mount point and return a list of util.PID to be processed
-func ListPIDs() ([]libpf.PID, error) {
-	pids := make([]libpf.PID, 0)
-	files, err := os.ReadDir(defaultMountPoint)
+func ListPIDs(callback func(pid libpf.PID) error) error {
+	dir, err := os.Open(defaultMountPoint)
 	if err != nil {
-		return nil, err
+		return err
 	}
-	for _, f := range files {
-		// Make sure this is a PID file entry
-		if !f.IsDir() {
```

Comment:
We don't expect directories that contain of numbers which translate to uint32 (PID range). So this case is indirectly caught be the following lines.